### ISSUE TITLE
Add a cookie manager for complex objects

### ISSUE THREAD

**Issue by kjeske** [hydrostack/hydro#72]
Add a functionality to store/read complex objects in cookies.

To implement:
```c#
T HydroComponent.Cookies.Get<T>(string key)
```

```c#
HydroComponent.Cookies.Set<T>(string key, T value)
```

Value will be serialized and encrypted (optionally?)

**Comment by taublast** [comment]
It will be, indeed :)

```csharp
public interface ICookieManager
{
   public T Get<T>(string key, bool secure = false, T defaultValue = default);
   public void Set<T>(string key, T value, bool secure = false, TimeSpan? expires = null);
   public void Set<T>(string key, T value, CookieOptions options);
   public void Delete(string key);
}
```

More: https://github.com/hydrostack/hydro/pull/74

**Comment by kjeske** [comment]
Fixed in #104
